hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PartitionPredicateImpl.java

```diff
@@ -49,23 +57,43 @@ public PartitionPredicateImpl() {
     /**
      * Creates a new PartitionPredicate.
      *
-     * @param partitionKey the partition key
+     * @param partitionKeys the partition keys
      * @param target       the target {@link Predicate}
      * @throws NullPointerException     if partitionKey or target predicate is {@code null}
      */
-    public PartitionPredicateImpl(Object partitionKey, Predicate<K, V> target) {
-        this.partitionKey = checkNotNull(partitionKey, "partitionKey can't be null");
+    public PartitionPredicateImpl(Set<? extends Object> partitionKeys, Predicate<K, V> target) {
+        this.partitionKeys = checkNotNull(partitionKeys, "partitionKeys can't be null");
+        checkTrue(partitionKeys.size() > 0, "partitionKeys must not be empty");
```

Comment:
I haven't included a backwards compatible constructor here, I can add one if this is a class that is directly interacted with by external consumers

Reply:
it is under impl package, there is no expected public usage.